rules/no-function-iterator.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const iteratorMethods = [
+	'map',
+	'forEach',
+	'every',
+	'filter',
+	'find',
+	'findIndex',
+	'some'
+];
+
+const isIteratorMethod = node => node.callee.property && iteratorMethods.indexOf(node.callee.property.name) !== -1;
+const hasFunctionArgument = node => node.arguments.length === 1 && (node.arguments[0].type === 'Identifier' || node.arguments[0].type === 'CallExpression');
+
+const create = context => ({
+	CallExpression: node => {
+		const sourcecode = context.getSourceCode();
```

Comment:
Getting the source code is expensive. We should only do it when we actually need it.